pyqg/errors.py

```diff
@@ -0,0 +1,10 @@
+class PYQGError(Exception):
+    """Base class for all PYQG exceptions"""
+    pass
+
+class DiagnosticNotFilledError(PYQGError):
+    """Raise this when trying access a diagnostic that has not been filled"""
+    def __init__(self, dname):
+        self.dname = dname
+        super().__init__(self.dname)
```

Comment:
There are two ways you could do this. The one you have now is just passing a raw message

```python
class DiagnosticNotFilledError(PYQGError):
    def __init__(self, message):
        super().__init__(message)
```python

OR

```python
class DiagnosticNotFilledError(PYQGError):
    def __init__(self, dname):
        message = f"Diagnostic {dname} has not been filled. Please adjust tavestart and taveint."
        super().__init__(message)
```

Reply:
for the second example, when raising the error, you would just do

```python
dname = "enstrophy_flux"
raise DiagnosticNotFilledError(dname)
# -> "Diagnostic enstrophy_flux has not been filled. Please adjust tavestart and taveint."
```